devicedefender/tests/DeviceDefenderTest.cpp

```diff
@@ -141,14 +130,147 @@ static int s_TestDeviceDefenderFailedTest(Aws::Crt::Allocator *allocator, void *
         ASSERT_INT_EQUALS(AWS_ERROR_IOTDEVICE_DEFENDER_UNSUPPORTED_REPORT_FORMAT, task->LastError());
         ASSERT_FAILS(task->StartTask());
         ASSERT_TRUE(aws_last_error() == AWS_ERROR_INVALID_STATE);
+    }
+
+    return AWS_ERROR_SUCCESS;
+}
+
+AWS_TEST_CASE(DeviceDefenderFailedTest, s_TestDeviceDefenderFailedTest)
+
+static int s_TestMqtt5DeviceDefenderResourceSafety(Aws::Crt::Allocator *allocator, void *ctx)
+{
+    (void)ctx;
+    {
```

Comment:
nit: This new scope contains the whole function body. Is it really required here?